avocado/core/status/repo.py

```diff
@@ -52,10 +52,15 @@ def _set_task_data(self, message):
             self._all_data[task_id] = []
         self._all_data[task_id].append(message)
 
-    def get_task_data(self, task_id):
+    def get_all_task_data(self, task_id):
         """Returns all data on a given task, by its ID."""
         return self._all_data.get(task_id)
 
+    def get_task_data(self, task_id, index):
+        """Returns data on a given task, by its ID."""
```

Comment:
I think we can make this docstring more clear. I thought it a bit confusing when I read the docstring of the previous method and this method.

What do you think about: "Returns the data on the index of a given task, by its ID."